plugins/woocommerce/src/Admin/API/Reports/Orders/DataStore.php

```diff
@@ -534,6 +542,52 @@ protected function get_coupons_by_order_ids( $order_ids ) {
 		return $coupons;
 	}
 
+	/**
+	 * Get order attributions data from order IDs.
+	 *
+	 * @param array $order_ids Array of order IDs.
+	 * @return array
+	 */
+	protected function get_order_attributions_by_order_ids( $order_ids ) {
+		global $wpdb;
+		$order_meta_table   = OrdersTableDataStore::get_meta_table_name();
+		$included_order_ids = implode( ',', $order_ids );
+
+		if ( OrderUtil::custom_orders_table_usage_is_enabled() ) {
+			/* phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared */
+			$order_attributions_meta = $wpdb->get_results(
+				"SELECT order_id, meta_key, meta_value
+					FROM $order_meta_table
+					WHERE order_id IN ({$included_order_ids})
+					AND meta_key IN ('_wc_order_attribution_source_type', '_wc_order_attribution_utm_source')
+					",
+				ARRAY_A
+			);
+			/* phpcs:enable */
+		} else {
+			/* phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared */
+			$order_attributions_meta = $wpdb->get_results(
+				"SELECT post_id as order_id, meta_key, meta_value
+					FROM {$wpdb->prefix}postmeta
+					WHERE post_id IN ({$included_order_ids})
+					AND meta_key IN ('_wc_order_attribution_source_type', '_wc_order_attribution_utm_source')
+					",
+				ARRAY_A
+			);
+			/* phpcs:enable */
+		}
+
+		$order_attributions = array();
+		foreach ( $order_attributions_meta as $meta ) {
+			if ( ! isset( $order_attributions[ $meta['order_id'] ] ) ) {
+				$order_attributions[ $meta['order_id'] ] = array();
+			}
+			$order_attributions[ $meta['order_id'] ][ $meta['meta_key'] ] = $meta['meta_value'];
+		}
+
+		return $order_attributions;
+	}
+
```

Comment:
📜 

I took quite a bit of time to analyse this approach. I was inclined to use `wc_get_orders` instead of direct SQL. The advantages of using `wc_get_orders` are that we don't have to worry about `HPOS` compatibility, we have the `WC_Order` object giving us more information to work with, and our code is a bit cleaner. 

I also reached out to @vedanshujain and asked his opinion. He said both approaches are acceptable.

I did a simple benchmark test using `microtime`, and I can see using direct SQL is slightly faster than `wc_get_orders`. In the context of analytics, I am inclined towards faster results than cleaner code. Let's move forward with this approach and we can tweak it if required at a latest stage.